From: =?utf-8?q?St=C3=A9phane_Graber?= <stephane.graber@canonical.com>
Date: Thu, 12 May 2016 22:44:06 +0800
Subject: Update dnsmasq parameters

Disable caching since it's a potential security issue (local dns cache poisoning).

See also: https://blueprints.launchpad.net/ubuntu/+spec/foundations-p-dns-resolving
---
 src/dns/nm-dns-dnsmasq.c | 4 ++--
 src/dns/nm-dns-manager.c | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/dns/nm-dns-dnsmasq.c b/src/dns/nm-dns-dnsmasq.c
index cb311df..74f426d 100644
--- a/src/dns/nm-dns-dnsmasq.c
+++ b/src/dns/nm-dns-dnsmasq.c
@@ -500,8 +500,8 @@ start_dnsmasq (NMDnsDnsmasq *self)
 	argv[idx++] = "--no-hosts"; /* don't use /etc/hosts to resolve */
 	argv[idx++] = "--bind-interfaces";
 	argv[idx++] = "--pid-file=" PIDFILE;
-	argv[idx++] = "--listen-address=127.0.0.1"; /* Should work for both 4 and 6 */
-	argv[idx++] = "--cache-size=400";
+	argv[idx++] = "--listen-address=127.0.1.1"; /* Should work for both 4 and 6 */
+	argv[idx++] = "--cache-size=0";
 	argv[idx++] = "--clear-on-reload"; /* clear cache when dns server changes */
 	argv[idx++] = "--conf-file=/dev/null"; /* avoid loading /etc/dnsmasq.conf */
 	argv[idx++] = "--proxy-dnssec"; /* Allow DNSSEC to pass through */
diff --git a/src/dns/nm-dns-manager.c b/src/dns/nm-dns-manager.c
index 75f8a10..c87b6db 100644
--- a/src/dns/nm-dns-manager.c
+++ b/src/dns/nm-dns-manager.c
@@ -1162,14 +1162,14 @@ update_dns (NMDnsManager *self,
 		;
 	}
 
-	/* If caching was successful, we only send 127.0.0.1 to /etc/resolv.conf
+	/* If caching was successful, we only send 127.0.1.1 to /etc/resolv.conf
 	 * to ensure that the glibc resolver doesn't try to round-robin nameservers,
 	 * but only uses the local caching nameserver.
 	 */
 	if (caching) {
 		g_strfreev (nameservers);
 		nameservers = g_new0 (char*, 2);
-		nameservers[0] = g_strdup ("127.0.0.1");
+		nameservers[0] = g_strdup ("127.0.1.1");
 	}
 
 	if (update) {
@@ -1549,7 +1549,7 @@ nm_dns_manager_stop (NMDnsManager *self)
 	_LOGT ("stopping...");
 
 	/* If we're quitting, leave a valid resolv.conf in place, not one
-	 * pointing to 127.0.0.1 if any plugins were active.  Thus update
+	 * pointing to 127.0.1.1 if any plugins were active.  Thus update
 	 * DNS after disposing of all plugins.  But if we haven't done any
 	 * DNS updates yet, there's no reason to touch resolv.conf on shutdown.
 	 */
